### ISSUE TITLE
poisson_gene_selection

### ISSUE THREAD

**Issue by xianjieshen** [scverse/scvi-tools#2399]
When running your tutorial, I encountered some errors. It is indicating that my 'anndata' is not recognized as raw data. However, my 'anndata' file contains raw data, and even when I use the raw matrix, it still doesn't work. Here are the details of the issue:
ValueError when using` poisson_gene_selection`
When attempting to use the `poisson_gene_selection function` from `scvi-tools`, a ValueError is raised indicating that 'raw count data' is expected. This occurs even when a dataset is loaded that I believe to be in the correct format.

```
import os
import tempfile
import warnings

import matplotlib.pyplot as plt
import numpy as np
import pandas as pd
import plotnine as p9
import scanpy as sc
import scvi
import torch
scvi.settings.seed = 0
print("Last run with scvi-tools version:", scvi.__version__)
adata=sc.read('/home/xjs/h5ad/adata.h5ad')
scvi.data.poisson_gene_selection(adata)
adata.var.head()
```

```
Global seed set to 0
/home/bachelortwo/miniconda3/envs/scanpy/lib/python3.9/site-packages/anndata/__init__.py:51: FutureWarning: `anndata.read` is deprecated, use `anndata.read_h5ad` instead. `ad.read` will be removed in mid 2024.
/home/bachelortwo/miniconda3/envs/scanpy/lib/python3.9/site-packages/anndata/_core/anndata.py:1906: UserWarning: Observation names are not unique. To make them unique, call `.obs_names_make_unique`.
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
Cell In[18], line 15
     13 print("Last run with scvi-tools version:", scvi.__version__)
     14 adata=sc.read('/home/xjs/h5ad/adata.h5ad')
---> 15 scvi.data.poisson_gene_selection(adata)
     16 adata.var.head()

File ~/miniconda3/envs/scanpy/lib/python3.9/site-packages/torch/utils/_contextlib.py:115, in context_decorator.<locals>.decorate_context(*args, **kwargs)
    112 @functools.wraps(func)
    113 def decorate_context(*args, **kwargs):
    114     with ctx_factory():
--> 115         return func(*args, **kwargs)

File ~/miniconda3/envs/scanpy/lib/python3.9/site-packages/scvi/data/_preprocessing.py:98, in poisson_gene_selection(adata, layer, n_top_genes, use_gpu, accelerator, device, subset, inplace, n_samples, batch_key, silent, minibatch_size)
     96 data = adata.layers[layer] if layer is not None else adata.X
     97 if _check_nonnegative_integers(data) is False:
---> 98     raise ValueError("`poisson_gene_selection` expects " "raw count data.")
    100 _, _, device = parse_device_args(
    101     use_gpu=use_gpu,
    102     accelerator=accelerator,
   (...)
    105     validate_single_device=True,
    106 )
    108 if batch_key is None:

ValueError: `poisson_gene_selection` expects raw count data.
```

#### Versions:

> VERSION

scVI 1.0.4

Thank you

**Comment by xianjieshen** [comment]
> In which layer is your raw count data? If it's not `adata.X`, then please pass in the `layer` argument into the function. Could you also run `np.all(a >= 0) & np.all(a % 1 == 0)` on your array to double-check?

My raw data has undergone[`Deepimpute`](https://github.com/lanagarmire/DeepImpute). 
Should there be zero values in this context?After imputation, can the function not be used if zero values have been filled?

**Comment by martinkim0** [comment]
I'm not familiar with DeepImpute, but the only restriction we put on this function is that values need to be non-negative integers. Are the zero values filled with non-integer numbers? If so, it's probably more suitable to use the pre-imputation raw counts.

**Comment by xianjieshen** [comment]
> I'm not familiar with DeepImpute, but the only restriction we put on this function is that values need to be non-negative integers. Are the zero values filled with non-integer numbers? If so, it's probably more suitable to use the pre-imputation raw counts.

I understand, thank you ! You've resolved my question.

**Comment by martinkim0** [comment]
I also realized that the error message could be more descriptive instead of just saying "raw count data". I'll go ahead and update this.

**Comment by martinkim0** [comment]
Addressed in #2422
